src/Roslyn.Diagnostics.Analyzers/Core/NamedTypeFullNameNotNullSuppressor.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Roslyn.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class NamedTypeFullNameNotNullSuppressor : DiagnosticSuppressor
+    {
+        private const string Id = RoslynDiagnosticIds.NamedTypeFullNameNotNullSuppressionRuleId;
+
+        // CS8600: Converting null literal or possible null value to non-nullable type
+        private const string CS8600 = nameof(CS8600);
+
+        // CS8603: Possible null reference return
+        private const string CS8603 = nameof(CS8603);
+
+        // CS8604: Possible null reference argument for parameter 'name' in 'method'
+        private const string CS8604 = nameof(CS8604);
+
+        private static readonly LocalizableString s_localizableJustification = new LocalizableResourceString(nameof(RoslynDiagnosticsAnalyzersResources.NamedTypeFullNameNotNullSuppressorJustification), RoslynDiagnosticsAnalyzersResources.ResourceManager, typeof(RoslynDiagnosticsAnalyzersResources));
+
+        internal static readonly SuppressionDescriptor CS8600Rule = new SuppressionDescriptor(Id, CS8600, s_localizableJustification);
+        internal static readonly SuppressionDescriptor CS8603Rule = new SuppressionDescriptor(Id, CS8603, s_localizableJustification);
+        internal static readonly SuppressionDescriptor CS8604Rule = new SuppressionDescriptor(Id, CS8604, s_localizableJustification);
+
+        public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions { get; } = ImmutableArray.Create(CS8600Rule, CS8603Rule, CS8604Rule);
+
+        public override void ReportSuppressions(SuppressionAnalysisContext context)
+        {
+            foreach (var diagnostic in context.ReportedDiagnostics)
+            {
+                if (!(diagnostic.Location.SourceTree is { } tree))
+                {
+                    continue;
+                }
+
+                var root = tree.GetRoot(context.CancellationToken);
+                var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                var semanticModel = context.GetSemanticModel(tree);
+                var operation = semanticModel.GetOperation(node, context.CancellationToken);
+                if (operation is IPropertyReferenceOperation { Property: { Name: nameof(Type.FullName) }, Instance: ITypeOfOperation { } })
```

Comment:
Is there a plan to change the compiler to detect this and avoid firing false positives? Why would such a suppression be specific to consumers of Roslyn.Diagnostics.Analyzers as opposed to everyone that enables nullable reference type features.

Reply:
Tagging @jcouv

Reply:
> Is there a plan to change the compiler to detect this and avoid firing false positives?

Not that I'm aware of. According to the signatures in the API, the compiler is correct in reporting the diagnostic.

> Why would such a suppression be specific to consumers of Roslyn.Diagnostics.Analyzers as opposed to everyone that enables nullable reference type features?

It's an experiment in using suppressions to target common patterns that can't be covered by the NRT feature itself. If it proves valuable and other users ask for it, we could move it to one of the shared packages.

Reply:
> According to the signatures in the API, the compiler is correct in reporting the diagnostic.

And we lack the expressivity necessary to differentiate these cases in the API. FullName can return null, it just won't with this calling pattern.

Reply:
Ah, thanks for the explanations. I still hope all customers eventually get benefit of these suppressions by default, not just the ones using Roslyn.Diagnostics.Analyzers.

Reply:
@stephentoub How should we track proposals to roll these out more generally?

Reply:
> Is there a plan to change the compiler to detect this and avoid firing false positives?

Hard to have a plan when we're not really involved in the discussions.